src/software/world/field.h

```diff
@@ -263,6 +263,13 @@ class Field
      */
     Rectangle enemyGoal() const;
 
+    /**
+     * Gets the area within the friendly goalie's no-chip rectangle
+     *
+     * @return the area within the friendly goalie's no-chip rectangle
+     */
+     Rectangle noChipRectangle() const;
```

Comment:
since noChipRectangle is only needed by the friendly goalie, why not have this function in goalie tactic fsm?

Reply:
Good point, I moved it here to keep it with the rest of the field stuff, but it also makes sense just in the fsm, I can move it back!